src/Aspire.Dashboard/Otlp/WebApplicationExtensions.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard.Otlp.Grpc;
+using Microsoft.AspNetCore.Mvc;
+using OpenTelemetry.Proto.Collector.Logs.V1;
+using OpenTelemetry.Proto.Collector.Metrics.V1;
+using OpenTelemetry.Proto.Collector.Trace.V1;
+
+namespace Aspire.Dashboard.Otlp;
+
+public static class WebApplicationExtensions
+{
+    public static WebApplication ConfigureHttpOtlp(this WebApplication app, Uri httpEndpoint)
+    {
+        app.MapPost("/v1/metrics",
+            async ([FromServices] OtlpMetricsService service, HttpContext httpContext) =>
+            {
+                var body = await ReadFullyAsync(httpContext).ConfigureAwait(false);
+                var metrics = ExportMetricsServiceRequest.Parser.ParseFrom(body);
+                await service.Export(metrics, null!).ConfigureAwait(false);
+            }
+        ).AllowAnonymous()
+        .RequireHost($"*:{httpEndpoint.Port}");
+
+        app.MapPost("/v1/traces",
+                async ([FromServices] OtlpTraceService service, HttpContext httpContext) =>
+                {
+                    var body = await ReadFullyAsync(httpContext).ConfigureAwait(false);
+                    var traces = ExportTraceServiceRequest.Parser.ParseFrom(body);
+                    await service.Export(traces, null!).ConfigureAwait(false);
+                }
+            ).AllowAnonymous()
+            .RequireHost($"*:{httpEndpoint.Port}");
+
+        app.MapPost("/v1/logs",
+            async ([FromServices] OtlpLogsService service, HttpContext httpContext) =>
+            {
+                var body = await ReadFullyAsync(httpContext).ConfigureAwait(false);
+                var logs = ExportLogsServiceRequest.Parser.ParseFrom(body);
+                await service.Export(logs, null!).ConfigureAwait(false);
+            }
+        ).AllowAnonymous()
+        .RequireHost($"*:{httpEndpoint.Port}");
+
+        return app;
+    }
+
+    private static async Task<byte[]> ReadFullyAsync(HttpContext context)
+    {
+        using var ms = new MemoryStream();
+        await context.Request.Body.CopyToAsync(ms).ConfigureAwait(false);
+        return ms.ToArray();
+    }
```

Comment:
This should be using pipelines to buffer without allocating.

Reply:
How do I achieve this?
`Google.Protobuf.MessageParser<T>` wants either `ReadOnlySequence<byte> `, `ReadOnlySpan<byte>` or a `Stream`. but the HttpContext.Request.Body-Stream can not be passed directly

Reply:
like this?
```
var reader = httpContext.Request.BodyReader;
var readResult = await reader.ReadAsync().ConfigureAwait(false);
var readResultBuffer = readResult.Buffer;
var metrics = ExportMetricsServiceRequest.Parser.ParseFrom(readResultBuffer);
```

Reply:
@davidfowl I got rid of the byte[] and stream allocation